backend/prisma/schema.prisma

```diff
@@ -32,8 +50,100 @@ model Session {
 }
 
 model Profile {
-  id     String @id @default(uuid())
-  user   User @relation(fields: [userId], references: [id])
-  userId String @unique @default(uuid())
-  photoUrl String? 
-}
\ No newline at end of file
+  profileID String  @id @default(uuid())
+  user      User    @relation(fields: [userId], references: [UserID])
+  userId    String  @unique
+  photoUrl  String?
+}
+
+model Plan {
+  PlanID           String      @id @default(uuid())
+  description      String
+  startDate        DateTime
+  endDate          DateTime
+  createdDate      DateTime    @default(now())
+  specialEventType EventType
+  contact          Contact?    @relation(fields: [contactID], references: [ContactID])
+  contactID        String?     @unique
+  User             Pledge[]
+  FinalGift        FinalGift?
+}
+
+model Pledge {
+  PledgeID     String        @id @default(uuid())
+  plan         Plan          @relation(fields: [planID], references: [PlanID])
+  planID       String
+  user         User          @relation(fields: [userID], references: [UserID])
+  userID       String
+  inviteStatus InviteStatus?
+  role         Role
+  giftChoice   GiftChoice[]
```

Comment:
should a pledge be tied to a gift or to a plan?

Reply:
My reasoning is that GiftID can be uniquely identify by the pledgeID, and in turn, the pledgeID is uniquely defined by the planID and userID. I think gift choices, like casting vote for a gift in a plan, should be stored with the pledgeID and not the planID.

If we do decide store the gift choices in the plan table, we'd have to make another join table between the Plan and User called 'UserGiftChoice' which is essentially the 'pledge' table rn.

Reply:
not sure if im just reading your comment wrong but the way I'm thinking about this is essentially that there is a 1 to many relationship between Plans and Pledges, or that a plan can have many pledges, but a pledge can have only one plan. this is represented correctly in your schema I believe

the question I'm asking essentially is should the pledge _have_ to be tied to a gift? consider the following sequence of events:
1) Plan leader chooses a laptop as the gift
2) I pledge $50 towards the laptop and submit $50
3) plan leader decides to switch to a ps5 instead

do we have to update the giftchoice for every pledge every time the gift is changed? that seems like unnecessary extra work to me, when we could just link a pledge strictly to a Plan and then not worry ourselves about the gift, since the Plan acts as the central linking entity

Reply:
Added gift choice table